models/stock_picking_batch.py

```diff
@@ -237,3 +237,81 @@ def is_valid_location_dest_id(self, location_ref):
 
         return all([pick.is_valid_location_dest_id(location=location)
                     for pick in all_done_pickings])
+
+    @api.multi
+    def unpickable_item(self, move_line_id, reason, picking_type_id=None):
+        """
+        Given a valid move_line_id create a new picking of type picking_type_id
+        picking for it. If it is the last move_line_id of the wave,
+        the wave is set as done.
+
+        The move_line_id is valid if it is in the current wave (self)
+        and its picking is not done or cancel.
+        """
+
+        self.ensure_one()
+
+        ResUsers = self.env['res.users']
+        StockMoveLine = self.env['stock.move.line']
+        Picking = self.env['stock.picking']
+        Group = self.env['procurement.group']
+
+        move_line = StockMoveLine.browse(move_line_id)
+        if not move_line.exists():
+            # We need to check these here lest we throw exceptions
+            # when getting picking_id, package_id etc
+            raise ValidationError(_('Cannot find the operation'))
+
+        picking = move_line.picking_id
+        package = move_line.package_id
+        location = move_line.location_id
+
+        if picking_type_id is None:
+            picking_type_id = ResUsers.get_user_warehouse().int_type_id.id
+        if picking.batch_id != self:
+            raise ValidationError(_('Move line is not part of the batch.'))
+        if picking.state in ['cancel', 'done']:
+            raise ValidationError(_('Cannot mark a move line as unpickable '
+                                    'when it is part of a completed Picking.'))
+
+        if package:
+            # This move line is part of a package.  Get all quants that are
+            # also on this package as well as all other move lines in the
+            # package as they will have been affected as well.
+            quants = package._get_contained_quants()
+            move_lines = picking.move_line_ids.get_package_move_lines(package)
+            msg = 'Unpickable package %s at location %s' % (package.name,
```

Comment:
`msg` should be wihtout the package name and location name and then add them at line 285 something like:
`picking.message_post(body=_(msg)% (package.name, location.name))
Because _ is for translation and the package and location names shouldn't be translated.